client/src/components/onboarding/UserAvailability.js

```diff
@@ -158,15 +160,69 @@ function UserAvailability({setters}) {
     const history = useHistory();
 
     const clickHandler = () => {
-        history.push("/home")
-        alertContext.setAlertStatus({
-            isOpen:true,
-            message:"Profile created!",
-            type:"success"
-            })    
-    }
-
 
+        const availableDays = days.filter(day => day.state)
+
+        let status=0;
+
+        try {
+            if (startTime === "" || endTime === "") {
+                throw Error("Available time cannot be empty")  
+            }
+            else if (startTime >= endTime) {
+                throw Error("Start time must be earlier than end time")    
+            } 
+            else if (availableDays.length === 0) {
+                throw Error("Must select at least one available date")    
+            }
```

Comment:
Oh, this is great :thumbsup: , I was thinking about the frontend checking when I was building BE API. 
Also, I think there should be an email format checking for other pages needs, I'll see if I got time to do it later.
For convenience, here is the regex for email: `^(\w|\.|\_|\-)+[@](\w|\_|\-|\.)+[.]\w{2,3}$`, and I've already added it in Appointment creation API.
And I think we need both frontend and backend checking to prevent hacking stuff.

Reply:
sorry where does the email validation occur? Aren't we using the user's Gmail?

Reply:
oh, guess the only one will be booking appointment then. I'll try to add it on later